src/components/DetailsMetrics/DetailsMetrics.js

```diff
@@ -122,13 +188,61 @@ const DetailsMetrics = ({ selectedItem }) => {
     )
   }, [dispatch, selectedItem])
 
+  useEffect(() => {
+    const selectedDate = detailsStore.dates.selectedOptionId
+    if (!selectedDate || !(selectedDate in timeRangeMapping)) return
+
+    setSelectedDate(timeRangeMapping[selectedDate])
+  }, [detailsStore.dates.selectedOptionId])
+
+  const fetchData = useCallback(
+    (params, params2, selectedItem) => {
+      metricsValuesAbortController.current = new AbortController()
+
+      return Promise.all([
+        dispatch(
+          detailsActions.fetchModelEndpointMetricsValues(
+            selectedItem.metadata.project,
+            selectedItem.metadata.uid,
+            params,
+            metricsValuesAbortController.current.signal
+          )
+        ),
+        dispatch(
+          detailsActions.fetchModelEndpointMetricsValues(
+            selectedItem.metadata.project,
+            selectedItem.metadata.uid,
+            params2,
+            metricsValuesAbortController.current.signal
+          )
+        )
+      ])
+        .then(([metrics, previousInvocation]) => {
+          setMetrics(metrics)
+
+          if (previousInvocation.length !== 0 && previousInvocation[0].data === true) {
+            setPreviousTotalInvocation(previousInvocation[0].rawDataTotal)
+          }
+        })
+        .catch(error => {
+          console.error(error.message)
+        })
+    },
+    [dispatch, setMetrics, metricsValuesAbortController]
+  )
+
   useEffect(() => {
     if (
       selectedItem.metadata?.uid &&
       !isEmpty(detailsStore.metricsOptions.selectedByEndpoint[selectedItem.metadata?.uid])
     ) {
       const selectedMetrics =
         detailsStore.metricsOptions.selectedByEndpoint[selectedItem.metadata?.uid]
+
+      const invocationMetric = detailsStore.metricsOptions.all.filter(
+        metric => metric.app === 'mlrun-infra'
+      )
```

Comment:
why you always use `filter` when you need only one item? You need to use `find`

Reply:
done.